resources/keycloak/custom-user-provider/src/main/java/com/cvmanager/auth/provider/user/UserAdapter.java

```diff
@@ -0,0 +1,249 @@
+            case UserModel.LAST_NAME:
+                return entity.getLastName();
+            case "created_timestamp":
+                return String.valueOf(entity.getCreatedTimestamp());
+            case "super_user":
+                return String.valueOf(entity.getSuperUser());
+            case "organizations":
+                return entity.getOrganizations() != null ? OrganizationObject.toStringList(entity.getOrganizations())
+                        : "[]";
+            default:
+                return super.getFirstAttribute(name);
+        }
+    }
+
+    @Override
+    public Map<String, List<String>> getAttributes() {
+        log.debug("getAttributes");
+        MultivaluedHashMap<String, String> attrs = new MultivaluedHashMap<>();
+        attrs.add(UserModel.USERNAME, entity.getEmail());
+        attrs.add(UserModel.EMAIL, entity.getEmail() != null ? entity.getEmail() : "");
+        attrs.add(UserModel.FIRST_NAME, entity.getFirstName() != null ? entity.getFirstName() : "");
+        attrs.add(UserModel.LAST_NAME, entity.getLastName() != null ? entity.getLastName() : "");
+        attrs.add("created_timestamp",
+                String.valueOf(entity.getCreatedTimestamp() != null ? entity.getCreatedTimestamp() : ""));
+        attrs.add("super_user",
+                String.valueOf(entity.getSuperUser() != null ? entity.getSuperUser() : ""));
+        attrs.add("organizations",
+                entity.getOrganizations() != null ? OrganizationObject.toStringList(entity.getOrganizations()) : "[]");
+        // Include other attributes as needed
+        return attrs;
+    }
+
+    @Override
+    public Stream<String> getAttributeStream(String name) {
+        log.debug("getAttributeStream: " + name);
+        switch (name) {
+            case UserModel.USERNAME:
+                return Stream.of(entity.getEmail());
+            case UserModel.EMAIL:
+                return entity.getEmail() != null ? Stream.of(entity.getEmail()) : Stream.empty();
+            case UserModel.FIRST_NAME:
+                return entity.getFirstName() != null ? Stream.of(entity.getFirstName()) : Stream.empty();
+            case UserModel.LAST_NAME:
+                return entity.getLastName() != null ? Stream.of(entity.getLastName()) : Stream.empty();
+            case "created_timestamp":
+                return entity.getCreatedTimestamp() != null ? Stream.of(String.valueOf(entity.getCreatedTimestamp()))
+                        : Stream.empty();
+            case "super_user":
+                return entity.getSuperUser() != null ? Stream.of(String.valueOf(entity.getSuperUser()))
+                        : Stream.empty();
+            case "organizations":
+                return entity.getOrganizations() != null
+                        ? Stream.of(OrganizationObject.toStringList(entity.getOrganizations()))
+                        : Stream.empty();
+            default:
+                return super.getAttributeStream(name);
+        }
```

Comment:
suggestion(non-blocking): this can be simplified by using the "enhanced switch statement", using `ofNullable` (which does what the same as what you're already doing, but in fewer characters), and merging the USERNAME and EMAIL cases since they're doing the same thing
```suggestion
        return switch (name) {
            case UserModel.USERNAME, UserModel.EMAIL -> Stream.ofNullable(entity.getEmail());
            case UserModel.FIRST_NAME -> Stream.ofNullable(entity.getFirstName());
            case UserModel.LAST_NAME -> Stream.ofNullable(entity.getLastName());
            case "created_timestamp" -> Stream.ofNullable(String.valueOf(entity.getCreatedTimestamp()));
            case "super_user" -> Stream.ofNullable(String.valueOf(entity.getSuperUser()));
            case "organizations" -> Stream.ofNullable(OrganizationObject.toStringList(entity.getOrganizations()));
            default -> super.getAttributeStream(name);
        };
```

Reply:
updated